eth_account/_utils/transaction_utils.py

```diff
@@ -0,0 +1,82 @@
+from typing import (
+    Any,
+    Dict,
+    Sequence,
+)
+
+from toolz import (
+    assoc,
+    dissoc,
+)
+
+from eth_account._utils.validation import (
+    is_rlp_structured_access_list,
+    is_rpc_structured_access_list,
+)
+
+
+def possibly_set_transaction_type(transaction_dict: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Hmm I checked it out. I think it might get a bit messy since we use this method in `legacy_transaction.py` as well before we can delegate to `TypedTransaction` based on the type. I could be missing something you are seeing though. Thoughts?

Reply:
Ah, no I forgot about the `legacy_transaction`. Never mind!